### ISSUE TITLE
Posterior duplicated along first axis when using InputTransform

### ISSUE THREAD

**Issue by DavidWalz** [pytorch/botorch#1078]
I'm seeing a strange behavior when using input transforms in SingleTaskGP.
```python
import torch
from botorch.models import SingleTaskGP
from botorch.fit import fit_gpytorch_model
from gpytorch.mlls import ExactMarginalLogLikelihood
from botorch.models.transforms.input import Normalize

X = torch.rand(5, 2)
Y = X**2

model = SingleTaskGP(X, Y, input_transform=Normalize(2))
mll = ExactMarginalLogLikelihood(model.likelihood, model)
fit_gpytorch_model(mll)

posterior = model.posterior(X)
```
The posterior mean and variance are duplicated along the first axis. 
```python
print(posterior.mean)
tensor([[[0.0419, 0.8879],
         [0.7072, 0.0928],
         [0.2388, 0.3047],
         [0.1625, 0.2043],
         [0.4918, 0.0091]],

        [[0.0419, 0.8879],
         [0.7072, 0.0928],
         [0.2388, 0.3047],
         [0.1625, 0.2043],
         [0.4918, 0.0091]]], grad_fn=<CopyBackwards>)
```
Without the transform the shape is (5, 2) as expected.

torch 1.10.2, gpytorch 1.6.0, botorch 0.6.0

**Comment by saitcakmak** [comment]
Thanks for flagging this! It is caused by the `_set_transformed_inputs` call introduced in #894. It calls the `input_transform` with `model.train_inputs` while it is still in `train` mode. Ordinarily, `model.train_inputs` would be (possibly reshaped) `train_X`, however, it seems that a batch dimension is added in `fit_gpytorch_model(mll)` due to each output of the multi-output `SingleTaskGP` being fitted independently and the model transforms that happen to facilitate that. Since the `Normalize` is not supplied with a `bounds` argument, it learns the `bounds` while in the `train` mode. So, the call with `model.train_inputs` ends up adding a batch dimension to the bounds, which is in turn added to `X` in `model.posterior(X)` leading to the batched output you see. I will put up a fix shortly.
